### PR TITLE
[FIX] 8.0 - base_user_role (regression)

### PR DESCRIPTION
**Pull request by sebalix**
This PR fix/replace the previous one (#922).
Ping @StefanRijnhart @novawish

### CODE REVIEW

**Review by StefanRijnhart — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
The tests are a mess, but I don't believe this module is too blame. All this module's tests succeeded.

**Comment by sebalix** [comment]
Yes, CI fails on `users_ldap_push`.

**Review by dreispt — APPROVED** [review]
LGTM

File: base_user_role/tests/test_user_role.py [hunk 1]
```diff
@@ -105,3 +105,61 @@ def test_role_1_2_with_dates(self):
...
+
+    def test_role_unlink(self):
+        # Get role1 groups
+        role1 = self.role_model.browse(self.cr, self.uid, self.role1_id)
+        role1_group_ids = role1.implied_ids.ids
+        role1_group_ids.append(role1.group_id.id)
+        role1_group_ids = sorted(set(role1_group_ids))
+        # Get role2 groups
+        role2 = self.role_model.browse(self.cr, self.uid, self.role2_id)
+        role2_group_ids = role2.implied_ids.ids
```
Comment (novawish) [inline base_user_role/tests/test_user_role.py]: I see that `role2_group_ids` was not used anywhere below, can we remove it?

File: base_user_role/tests/test_user_role.py [hunk 2]
```diff
@@ -105,3 +105,61 @@ def test_role_1_2_with_dates(self):
...
+            {'role_line_ids': [
+                (0, 0, {'role_id': self.role1_id}),
+                (0, 0, {'role_id': self.role2_id}),
+            ]})
+        user = self.user_model.browse(self.cr, self.uid, self.user_id)
+        # Remove role2
+        role2.unlink()
+        user_group_ids = sorted(set([group.id for group in user.groups_id]))
+        self.assertEqual(user_group_ids, role1_group_ids)
+        # Remove role1 from the user
```
Comment (novawish) [inline base_user_role/tests/test_user_role.py]: I think it should be `# Remove role1`, not from the user

**Review by novawish — COMMENTED** [review]
Just some minor comments, other than that LGTM

File: base_user_role/tests/test_user_role.py (hunk 1 above)
Comment (sebalix) [reply base_user_role/tests/test_user_role.py]: @novawish indeed, fixed

File: base_user_role/tests/test_user_role.py (hunk 2 above)
Comment (sebalix) [reply base_user_role/tests/test_user_role.py]: @novawish fixed

**Review by novawish — APPROVED** [review]

**Review by andhit-r — APPROVED** [review]

**Pull request merged**